vmsize_test.go

```diff
@@ -0,0 +1,192 @@
+		expectedVM: VMSizeType{
+			family:           "M",
+			subfamily:        nil,
+			cpus:             "8",
+			cpusConstrained:  stringToPtr("2"),
+			additiveFeatures: []rune{'m', 's'},
+			acceleratorType:  nil,
+			version:          "v2",
+			series:           "Mms_v2",
+		},
+	},
+	{
+		name: "Standard_A4_v2",
+		size: "A4_v2",
+		expectedVM: VMSizeType{
+			family:           "A",
+			subfamily:        nil,
+			cpus:             "4",
+			cpusConstrained:  nil,
+			additiveFeatures: []rune{},
+			acceleratorType:  nil,
+			version:          "v2",
+			series:           "A_v2",
+		},
+	},
+}
+
+// Test_getSubFamily tests the getSubFamily() function.
+func Test_getSubFamily(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetSubFamily()
+		subFamily, err := getSubFamily(test.size)
+		if err != nil {
+			t.Fatalf(`getSubFamily failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.subfamily, subFamily)
+	}
+}
+
+// Test_getConstrainedVCPUs tests the getSubFamily() function.
+func Test_getConstrainedVCPUs(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetConstrainedVCPUs()
+		constrainedVCPUs, err := getConstrainedVCPUs(test.size)
+		if err != nil {
+			t.Fatalf(`getConstrainedVCPUs failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.cpusConstrained, constrainedVCPUs)
+	}
+}
+
+// Test_getAdditiveFeatures tests the getAdditiveFeatures() function.
+func Test_getAdditiveFeatures(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetAdditiveFeatures()
+		additiveFeatures, err := getAdditiveFeatures(test.size)
+		if err != nil {
+			t.Fatalf(`getAdditiveFeatures failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.additiveFeatures, additiveFeatures)
+	}
+}
+
+// Test_getAcceleratorType tests the getAcceleratorType() function.
+func Test_getAcceleratorType(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetAcceleartorType()
+		acceleratorType, err := getAcceleratorType(test.size)
+		if err != nil {
+			t.Fatalf(`getAcceleratorType failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.acceleratorType, acceleratorType)
+	}
+}
+
+// Test_getVersion tests the getVersion() function.
+func Test_getVersion(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetVersion()
+		version, err := getVersion(test.size)
+		if err != nil {
+			t.Fatalf(`getVersion failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.version, version)
+	}
+}
+
+// Test_getSeries tests the getSeries() function.
+func Test_getSeries(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetSeries()
+		series, err := getSeries(test.size, test.expectedVM.family)
+		if err != nil {
+			t.Fatalf(`getSeries failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.series, series)
+	}
+}
+
+func stringToPtr(s string) *string {
+	return &s
+}
```

Comment:
https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/to/to.go#L9 fun part about generics is now we don't have to create XTypeToPtr
 
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
is the import, very useful library.

https://pkg.go.dev/github.com/samber/lo#ToPtr is another example of a generic pointer implementation with a much wider array of utilites as well. Its adopted throughout aws's karpenter project it seems.